h2/src/main/org/h2/expression/Function.java

```diff
@@ -1494,6 +1495,49 @@ private Value getValueWithArgs(Session session, Expression[] args) {
         case DATE_DIFF:
             result = ValueLong.get(datediff(v0.getString(), v1, v2));
             break;
+        case DATE_TRUNC:   
+            // Retrieve the field value (e.g. 'day', 'microseconds', etc.).
+            String fieldValue = v0.getString().toLowerCase();
```

Comment:
Do not use `String.toLowerCase()` without Locale argument. Some locales have own case conversion rules even for Latin letters.

It's may better to use `StringUtils.toUpperEnglish()` because this method maintains a cache and should work faster. (There is no such cache for `StringUtils.toLowerEnglish()`).